# HG changeset patch
# User Carlo de Falco <carlo.defalco@polimi.it>
# Date 1475080480 -7200
#      Wed Sep 28 18:34:40 2016 +0200
# Node ID 6b52d1e7ed56545e096229c8d40b03df0dd1540c
# Parent  dadcd4f21889976e4e7b212362d0ecddc20845a4
# Parent  29dab5b8332ac5c99f2b7421ce6f58d1f2f89386
maint: merged remote before pushing changeset

diff --git a/scripts/audio/@audioplayer/audioplayer.m b/scripts/audio/@audioplayer/audioplayer.m
--- a/scripts/audio/@audioplayer/audioplayer.m
+++ b/scripts/audio/@audioplayer/audioplayer.m
@@ -77,18 +77,18 @@
 ## play (player);
 ## # play for as long as you want
 ## stop (player);
 ## @end group
 
 function player = audioplayer (varargin)
 
   if (nargin < 1 || nargin > 4
-      || (nargin < 2 && (isa (varargin{1}, "function_handle")
-                         || ischar (varargin{1}))))
+      || (nargin < 2 && ! (is_function_handle (varargin{1})
+                           || ischar (varargin{1}))))
     print_usage ();
   endif
 
   if (isa (varargin{1}, "audiorecorder"))
     if (nargin == 1)
       player = getplayer (varargin{1});
     elseif (nargin == 2)
       recorder = varargin{1};
diff --git a/test/system.tst b/test/system.tst
--- a/test/system.tst
+++ b/test/system.tst
@@ -17,22 +17,21 @@
 ## <http://www.gnu.org/licenses/>.
 
 %!test
 %! [t1, u1, s1] = cputime ();
 %! for i = 1:200
 %!   sin (i);
 %! endfor
 %! [t2, u2, s2] = cputime ();
-%! assert (t1, u1 + s1);
-%! assert (t2 == u2 + s2);
+%! assert (t1, u1 + s1, 2*eps (t1));
+%! assert (t2, u2 + s2, 2*eps (t2));
 %! assert (t2 >= t1);
 %! assert (u2 >= u2);
 %! assert (s2 >= s2);
-%!#assert (t1 == u1 + s1 && t2 == u2 + s2 && t2 >= t1 && u2 >= u2 && s2 >= s2);
 
 %!test
 %! tic ();
 %! pause (2);
 %! assert (toc () > 0);
 
 %!test
 %! pause (0);
